Plugins/BuildServerIntegration/JenkinsIntegration/JenkinsAdapter.cs

```diff
@@ -118,54 +120,54 @@ public class JenkinsCacheInfo
             public long Timestamp = -1;
         }
 
-        private Task<ResponseInfo> GetBuildInfoTask(string projectUrl, bool fullInfo, CancellationToken cancellationToken)
+        private async Task<ResponseInfo> GetBuildInfoTaskAsync(string projectUrl, bool fullInfo, CancellationToken cancellationToken)
         {
-            return GetResponseAsync(FormatToGetJson(projectUrl, fullInfo), cancellationToken)
-                .ContinueWith(
-                    task =>
+            string t = null;
+            long timestamp = 0;
+            IEnumerable<JToken> s = Enumerable.Empty<JToken>();
+
+            try
+            {
+                t = await GetResponseAsync(FormatToGetJson(projectUrl, fullInfo), cancellationToken).ConfigureAwait(false);
+            }
+            catch
+            {
+                // Could be cancelled or failed
+            }
+
```

Comment:
cancellationToken.ThrowIfCancellationRequested();

Reply:
➡️ I am adding a comment to clarify that the current code has the intended semantics for this pull request.

Reply:
I think the comment is unnecessary. It was already clear. I prefer to be on the safe side here and ensure that there is no race condition here. Even if it was not implemented correctly before, then the intention for me is clear: do not process if the task is canceled.